accounts-journey/presentation/viewmodel/AccountDetailsViewModel.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  AccountDetailsViewModel.swift
+//  AccountsJourney
+//
+//  Created by Backbase R&D B.V. on 29/11/2023.
+//
+
+import Foundation
+import Resolver
+import BackbaseDesignSystem
+
+final class AccountDetailsViewModel: NSObject {
+    
+    // MARK: - Properties
+    @Published private(set) var screenState: AccountDetailsScreenState = .loading
+    
+    // MARK: - Private
+    private lazy var accountDetailsUseCase: AccountDetailsUseCase = {
+        guard let useCase = Resolver.optional(AccountDetailsUseCase.self) else {
+            fatalError("AccountDetailsUseCase needed to continue")
+        }
+        return useCase
+    }()
+    
+    // MARK: - Methods
+    func onEvent(_ event: AccountDetailsEvent) {
+        switch event {
+        case let .getAccountDetails(id):
+            getAccountDetail(id: id)
+        }
+    }
+    
+    func getAccountDetail(id: String) {
+        screenState = .loading
+        
+        accountDetailsUseCase.getAccountDetail(arrangementId: id) {[weak self] result in
+            guard let self else {
+                return
+            }
+            
+            switch result {
+            case let .success(accountDetailsResponse):
+                screenState = .loaded( accountDetailsResponse.toMapUI())
+            case let .failure(errorResponse):
+                screenState = .hasError(stateViewConfiguration(for: .loadingFailure(errorResponse), primaryAction: {
+                    self.onEvent(.getAccountDetails(id))
+                }))
+            }
+        }
+    }
+}
+
+public enum AccountDetailsEvent {
+    case getAccountDetails(String)
```

Comment:
Does this make sense if there is just one event?

Reply:
For now it doesn't, but when we'll need to add a refresh, or just an example of how to structure application events.